appium/lib/screen-object/accessors/element.rb

```diff
@@ -81,44 +81,119 @@ def dynamic_text_exists? dynamic_text
         end
       end
 
-      def scroll
-        $driver.execute_script 'mobile: scrollTo',:element => element.ref
-        # $driver.execute_script("mobile: scroll",:direction => direction.downcase, :element => element.ref)
+      def gesture(x1,y1,x2,y2,duration)
+        Appium::TouchAction.new($driver).swipe(start_x: x1, start_y: y1, end_x: x2, end_y: y2,duration: duration).perform
+      rescue
+        raise("Error during gesture")
       end
 
-      def scroll_to_text(text)
-        $driver.scroll_to(text)
+      def _scroll(direction = :down, duration = 1000)
+        size = driver.window_size
+        x = size.width/2
+        y = size.height/2
+        case direction
+        when :up then loc = [x, y*0.5, x, (y + (y*0.5)),duration]
+        when :down then loc =[x, y, x, y * 0.5,duration]
+        when :left then loc = [x * 0.6, y, x * 0.3, y,duration]
+        when :right then loc =[x * 0.3, y, x * 0.6, y,duration]
+        else
+          raise('Only upwards and downwards scrolling are supported')
+        end
+        gesture(loc[0],loc[1],loc[2],loc[3],loc[4])
```

Comment:
How about you just call `swipe(loc)` here and then update the `swipe` method you deleted to do `gesture(loc[0],loc[1],loc[2],loc[3],loc[4])`